### PR TITLE
Story/quad 116 sprint 5

### PR DESCRIPTION
**Pull request by rajataggarwal91**
This pull request consists of all the tasks in quad-90, quad-115 and quad-116. Front end and back end functionalities are implemented for edit about page for public websites.

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    @Id
+    private String Id;
+
+    public String getId() {
+        return Id;
+    }
+
+    public void setId(String id) {
+        Id = id;
+    }
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java]: all your getters/setters should be together and all your variable declarations should be together. these methods should be moved down.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/publicwebsite/IAboutTextDAO.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+public interface IAboutTextDAO extends IBaseDAO<AboutTextDTO> {
+
+    /**
+     * Saving "about project" information in AboutTextDTO object
+     * 
+     * @author Rajat Aggarwal
+     */
+    public abstract void saveNewDTO(AboutTextDTO dto);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/publicwebsite/IAboutTextDAO.java]: why did you add this method here? It's already in IBaseDAO, isn't it?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Basic(optional = false)
+    @Column(name = "ProjectId")
+    private String projectId;
+
+    @Basic(optional = false)
+    @Column(name = "title")
+    private String title;
+
+    @Basic(optional = false)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java]: I wouldn't put these restrictions on the database level. You can check that there is a title/description in the validation of the form, but I wouldn't put it here. This might be something we'd like to make optional in the future. Even with projectId I would be careful. It makes it more difficult to reuse in the future, if we put those kind of restrictions on the database level.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+@RequestMapping(value="auth/workbench/projects/{projectId}/settings")
+public String showSettings(@PathVariable("projectId") String projectId, Model model, Principal principal) throws QuadrigaStorageException{
+	
+	IProject project = projectManager.getProjectDetails(projectId);
+    model.addAttribute(project);
+	return "auth/settings";
+}
+	
+	
+}
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java]: formatting

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    @Autowired
+    private IAboutTextManager aboutTextManager;
+
+    @PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
+    @RequestMapping(value = "auth/workbench/projects/{projectId}/settings/saveabout", method = RequestMethod.POST)
+    public String saveAbout(@PathVariable("projectId") String projectId, @ModelAttribute("AboutTextBackingBean") AboutTextBackingBean formBean, Principal principal)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java]: this method should be in WebsiteAboutEditController. It handle the save requests sent from the edit page, right? As a general rule, you could say one controller per page (there are exceptions of course).

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java]: You also want to think about if there is something you want to validate before storing.

File: Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp [hunk 6]
```diff
@@ -7,4 +7,7 @@
 <li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a
         href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/search">Search</a></li>
 <li ${currentPage == "getprojectstatistics" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
\ No newline at end of file
+        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
+<li ${currentPage == "about" ? "class=\"active\"" : ""}><a
+	href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/about">About</a></li>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp]: About should be the second menu entry from the left

File: Quadriga/src/main/webapp/WEB-INF/views/settings/publicwebsitesettings.jsp [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<title>Public Page Settings</title>
+</head>
+<body>
+
+<a href="${pageContext.servletContext.contextPath}/auth/workbench/projects/${project.projectId}/settings/editabout">Change About Project details</a>
+
+</body>
+</html>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/settings/publicwebsitesettings.jsp]: This should not be a complete html page. no html or body tag needed.

File: Quadriga/src/main/webapp/WEB-INF/views/settings/saveAbout.jsp [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<title>Insert title here</title>
+</head>
+<body>
+Your information is successfully saved.
+</body>
+</html>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/settings/saveAbout.jsp]: same here

**Comment by jdamerow** [comment]
please reopen this PR when you're done

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java (hunk 5 above)
Comment (rajataggarwal91) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java]: I cannot think of anything right now to validate. I believe if a user wants to keep everything as blank, he/she should be able to. This is contrary to the checks I had put up in database earlier, but it makes more sense to me to keep it that way now. Please let me know if you have anything on your mind and I will implement that.

**Comment by rajataggarwal91** [comment]
All the above comments were addressed. Please review again.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/publicwebsite/IAboutTextDAO.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ */
+
+public interface IAboutTextDAO extends IBaseDAO<AboutTextDTO> {
+
+    /**
+     * Saving "about project" information in AboutTextDTO object
+     * 
+     * @author Rajat Aggarwal
+     */
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/publicwebsite/IAboutTextDAO.java]: delete the javadoc as well

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+	@Basic(optional = true)
+	@Column(name = "ProjectId")
+	private String projectId;
+
+	@Basic(optional = true)
+	@Column(name = "title")
+	private String title;
+
+	@Basic(optional = true)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dto/AboutTextDTO.java]: just remove the "Basic" annotation. That's the default anyways.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java (hunk 5 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutTextSaveController.java]: Hm, I think if the About page can be blank, then the link shouldn't show up in that case. It doesn't make much sense to have a link to a page that is blank. Also saving an empty object to the DB doesn't make much sense.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+     * 
+     * @author Rajat Aggarwal
+     *
+     */
+
+    @PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
+    @RequestMapping(value = "auth/workbench/projects/{projectId}/settings/saveabout", method = RequestMethod.POST)
+    public String saveAbout(@PathVariable("projectId") String projectId, @ModelAttribute("AboutTextBackingBean") AboutTextBackingBean formBean, Principal principal)
+            throws QuadrigaStorageException {
+        aboutTextManager.saveAbout(projectId, formBean.getTitle(), formBean.getDescription());
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: You are saving a new object here every time, don't you? This will result in several aboutTextDTOs for the same project.

**Comment by jdamerow** [comment]
see comments. Please reopen PR when done.

**Comment by rajataggarwal91** [comment]
Changes related to AboutTextDTO were made. Backend functionality is completed now, with edit about storing formatted output in database and about project public page reading from it. Tested for 2 projects.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+		return aboutTextDTO;
+	}
+
+	/**
+	 * Hibernate query to fetch any existing DTO object
+	 * 
+	 * @author: Rajat Aggarwal
+	 */
+
+	protected AboutTextDTO getPresentDTO(String projectId) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java]: what does this mean?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+	 * 
+	 * @param projectId
+	 *            Returns a DTO if already present for the given projectId
+	 * @return
+	 */
+
+	public AboutTextDTO getDTOByProjectId(String projectId) {
+
+		AboutTextDTO aboutTextDTO = getPresentDTO(projectId);
+		if (aboutTextDTO == null) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java]: business logic should go into service

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+		return aboutTextDTO.getTitle();
+
+	}
+
+	@Transactional
+	@Override
+	public String getAboutDescription(String projectId) {
+		AboutTextDTO aboutTextDTO = aboutTextDAO.getDTOByProjectId(projectId);
+		return aboutTextDTO.getDescription();
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java]: just have one method to get dto and use the dto's method to get title or description

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ */
+
+@Controller
+public class PublicWebsiteSettings {
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java]: nope, should use accesspolicy. this let's any quadriga admin or user call this method.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private IAboutTextManager aboutTextManager;
+
+	@PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
+	@RequestMapping(value = "auth/workbench/projects/{projectId}/settings/editabout", method = RequestMethod.GET)
+	public String showAbout(@PathVariable("projectId") String projectId, Model model, Principal principal)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: should be more specific

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+			throws QuadrigaStorageException {
+		IProject project = projectManager.getProjectDetails(projectId);
+
+		String title = null;
+		String description = null;
+
+		title = aboutTextManager.getAboutTitle(projectId);
+		description = aboutTextManager.getAboutDescription(projectId);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: same as before use object. and declare variables where they are used.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+		System.out.println(description);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: remove sysout

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+	 *
+	 */
+
+	@PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
+	@RequestMapping(value = "auth/workbench/projects/{projectId}/settings/saveabout", method = RequestMethod.POST)
+	public String saveAbout(@PathVariable("projectId") String projectId,
+			@ModelAttribute("AboutTextBackingBean") AboutTextBackingBean formBean, Principal principal)
+			throws QuadrigaStorageException {
+		aboutTextManager.saveAbout(projectId, formBean.getTitle(), formBean.getDescription());
+		return "auth/saveabout";
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: go back to edit page with a flash message saying saving was successful

**Comment by jdamerow** [comment]
please reopen when done

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java (hunk 19 above)
Comment (rajataggarwal91) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: So I should not have saveAbout page at all? Should I also pass a flag in controller which says if the text was saved or not? Because the first time the user goes to Edit About page I do not want to show the message or call saveAbout. Is there a better way to do it?

**Comment by rajataggarwal91** [comment]
All the changes specified in comments done now, including removal of saveAbout page.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java (hunk 15 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java]: This still needs to be fixed.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java [hunk 20]
```diff
@@ -0,0 +1,40 @@
+package edu.asu.spring.quadriga.web.publicwebsite;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicWebsiteSettings.java]: That's the wrong package. Only controllers serving the public webpages should go here. Just create a new subpackage in 'web' called 'settings' and put the class in there.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 21]
```diff
@@ -0,0 +1,75 @@
+package edu.asu.spring.quadriga.web.publicwebsite;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: same here. this class should go in the settings package.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+public class WebsiteAboutEditController {
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private IAboutTextManager aboutTextManager;
+
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 0, userRole = {
+			RoleNames.ROLE_PROJ_COLLABORATOR_EDITOR }) })
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: Project owners and project admins should be able to change the about text. Right now you let editors do it.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+
+	/**
+	 * . Any change made in the about project page is updated into the database
+	 * here and a "Successfully saved" message is displayed.
+	 * 
+	 * @author Rajat Aggarwal
+	 *
+	 */
+
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 0, userRole = {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: same as above

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+			RoleNames.ROLE_PROJ_COLLABORATOR_EDITOR }) })
+	@RequestMapping(value = "auth/workbench/projects/{projectId}/settings/saveabout", method = RequestMethod.POST)
+	public String saveAbout(@PathVariable("projectId") String projectId,
+			@ModelAttribute("AboutTextBackingBean") AboutTextBackingBean formBean, Model model, Principal principal)
+			throws QuadrigaStorageException {
+		aboutTextManager.saveAbout(projectId, formBean.getTitle(), formBean.getDescription());
+		IProject project = projectManager.getProjectDetails(projectId);
+		model.addAttribute("show_success_alert", true);
+		model.addAttribute("success_alert_msg", " Edit was successful.");
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutEditController.java]: Change this to: "You successfully edited the about text."

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+		aboutTextDTO.setProjectId(projectId);
+		aboutTextDTO.setTitle(title);
+		aboutTextDTO.setDescription(description);
+		aboutTextDAO.saveNewDTO(aboutTextDTO);
+	}
+
+	@Transactional
+	@Override
+	public AboutTextDTO getDTOByProjectId(String projectId) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java]: Nope. You manager can retrieve DTOs from the DAO, but it should not pass them on. You should return an interface here for a domain class. If you domain class is the same as the dto, you can have your dto implement the interface and still return the dto, but classes using the manager shouldn't know that they get DTOs.

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java]: Also, change the method name accordingly.

**Comment by rajataggarwal91** [comment]
Pushed changes according to comments. Please review.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+			Query query = sessionFactory.getCurrentSession().getNamedQuery("AboutTextDTO.findByProjectId");
+			query.setParameter("projectId", projectId);
+			List<AboutTextDTO> aboutTextDTOList = query.list();
+			if (aboutTextDTOList.size() != 0)
+				aboutTextDTO = aboutTextDTOList.get(0);
+			else
+				return null;
+		} catch (HibernateException e) {
+			System.out.println("Error while retrieving aboutTextDTO");
+			return null;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/publicwebsite/AboutTextDAO.java]: handle exception

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private IAboutTextManager aboutTextManager;
+
+	@RequestMapping(value = "sites/{ProjectUnixName}/about", method = RequestMethod.GET)
+	public String showAbout(@PathVariable("ProjectUnixName") String unixName, Model model, Principal principal)
+			throws QuadrigaStorageException {
+		IProject project = projectManager.getProjectDetailsByUnixName(unixName);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java]: merge develop and try using the new annotation to inject the project:
https://diging.atlassian.net/wiki/display/QUAD/Injecting+projects+by+unix+names

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+
+		aboutTextDTO.setProjectId(projectId);
+		aboutTextDTO.setTitle(title);
+		aboutTextDTO.setDescription(description);
+		aboutTextDAO.saveNewDTO(aboutTextDTO);
+	}
+
+	@Transactional
+	@Override
+	public IAboutText getDTOByProjectId(String projectId) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/publicwebsite/AboutTextManager.java]: name still needs to be adjusted

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/settings/PublicWebsiteSettings.java [hunk 29]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ */
+
+@Controller
+public class PublicWebsiteSettings {
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@PreAuthorize("hasRole('ROLE_QUADRIGA_USER_ADMIN') OR hasRole('ROLE_QUADRIGA_USER_STANDARD')")
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/settings/PublicWebsiteSettings.java]: still need to use the correct access controls here

**Comment by rajataggarwal91** [comment]
All above issues resolved. Also merged with develop. Please review.

**Comment by rajataggarwal91** [comment]
Closing this pull request and creating a pull request directly to develop
